### ISSUE TITLE
Refactor mp-ipfs queue

### ISSUE THREAD

**Issue by dariusc93** [Satellite-im/Warp#146]
Similar to #141, we would want to have a separate module that can check the connection status of each user with a request pending. The queue would contain an entry points for each key, that would have its own task performing the check and once connected to send the request over. Once the task is complete, it would signal the queue to remove the entry related to the user.
